### PR TITLE
Update Eigen version to commit e5e305a158a029f5b5f837bf821411a51439a970.

### PR DESCRIPTION
**Pull request by rmlarsen**
PiperOrigin-RevId: 201624024

### CODE REVIEW

**Review by gunan — APPROVED** [review]

File: tensorflow/workspace.bzl
Comment (gunan) [inline tensorflow/workspace.bzl]: I don't think we still need this patch. Could you try removing this line? (If you can, removing corresponding file will also be very helpful)

**Comment by wdirons** [comment]
This level of eigen contains a typo that fails the compile on ppc64le. Is it possible to not merge this into 1.9 until the typo is fixed? I can create a PR for the master branch when it is fixed.

The bug is this commit: https://bitbucket.org/eigen/eigen/commits/d2f8e1b572c0/
They are missing the closing ')'. I requested an account at eigen so I can comment on the bug. (Trying to get this fixed as soon as possible)

**Comment by rmlarsen** [comment]
@wdirons we really should have this change in release R1.9. It will be a major performance improvement. However, I can submit a PR to Eigen to fix the issue you mention first.

**Comment by wdirons** [comment]
Thank you @rmlarsen , I also got my team mate to add a comment to the bug: http://eigen.tuxfamily.org/bz/show_bug.cgi?id=1555

I'll keep an eye on it for a fix.

**Comment by rmlarsen** [comment]
Upstream Eigen PR: https://bitbucket.org/eigen/eigen/pull-requests/410/fix-typo-in-pbend-for-altivec/diff

**Comment by wdirons** [comment]
@rmlarsen , your patch was merged and I verified it on ppc64le. Thank you!

PR #20254 created to update Eigen in the master branch.

**Comment by case540** [comment]
Re-triggering Windows Bazel build for a third time. Not sure why it keeps on failing, but I feel its unlikely due to this change.

**Comment by gunan** [comment]
@case540 ready to merge when the tests pass.

**Pull request merged**
